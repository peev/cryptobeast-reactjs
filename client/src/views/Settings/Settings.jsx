// @flow
import React from 'react';
import { Grid } from '@material-ui/core';
import { inject, observer } from 'mobx-react';

import { RegularCard, ItemGrid } from './../../components';
import RegularButton from './../../components/CustomButtons/Button';
import CreatePortfolio from './../../components/Modal/CreatePortfolio';
import PortfoliosTable from './../../components/CustomTables/PortfoliosTable';
import TimeSettings from '../../components/Cards/TimeSettings';
import ApiIntegrations from './../../components/CustomTables/ApiIntegrations';
import AddApiAccount from './../../components/Modal/ApiAccountModals/AddApiAccount';
import NotificationSnackbar from '../../components/Modal/NotificationSnackbar';

import Weidex from './../../services/Weidex';


type Props = {
  MarketStore: {
    getSyncedSummaries: Function,
    getBaseCurrencies: Function,
  },
};

const Settings = inject('MarketStore')(observer(({ ...props }: Props) => {
  const getMarketSummaries = () => {
    props.MarketStore.getSyncedSummaries();
    props.MarketStore.getBaseCurrencies();
    props.MarketStore.syncMarketPriceHistory();
  };

  const syncData = () => {
    Weidex.sync({
      id: 'Test'
    });
    // TODO: Set loading or something
  };

  return (
    <Grid container>
      <ItemGrid xs={12} sm={12} md={12}>
        <RegularCard
          cardTitle="API Integrations"
          button={<AddApiAccount />}
          content={
            <ApiIntegrations
              tableHead={['Exchange', 'Account', 'Status', 'Key', 'Secret', '']}
            />
          }
        />
      </ItemGrid>

      <ItemGrid xs={12} sm={12} md={12} style={{ margin: '50px 0' }}>
        <RegularCard
          cardTitle="Portfolios"
          button={<CreatePortfolio place="settings" />}
          content={
            <PortfoliosTable
              tableHead={[
                { id: 'name', numberic: false, disablePadding: false, label: 'Name' },
                { id: 'numShares', numberic: true, disablePadding: false, label: 'Number of Shares' },
                { id: 'sharePrice', numberic: true, disablePadding: false, label: 'Current share price' },
                { id: 'totalUSD', numberic: true, disablePadding: false, label: 'Total Amount' },
                { id: 'update', numberic: false, disablePadding: false, label: '' },
                { id: 'delete', numberic: false, disablePadding: false, label: '' },
              ]}
            />
          }
        />
      </ItemGrid>

      <ItemGrid xs={12} sm={12} md={12} style={{ marginBottom: '20px' }}>
        <RegularCard
          cardTitle="Time Settings"
          content={<TimeSettings />}
        />
      </ItemGrid>

      <ItemGrid xs={12} sm={3} md={3}>
        <RegularButton color="primary" onClick={() => getMarketSummaries()}>
          Get Markets
        </RegularButton>
      </ItemGrid>

      <ItemGrid xs={12} sm={3} md={3}>
        <RegularButton color="primary" onClick={() => syncData()}>
          Sync data
        </RegularButton>
      </ItemGrid>

      <NotificationSnackbar />
    </Grid>
  );
}));

export default Settings;
