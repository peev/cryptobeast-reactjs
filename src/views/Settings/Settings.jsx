import React, { Component } from 'react';
import { Grid, Button, Snackbar } from 'material-ui';
import { RegularCard, ItemGrid } from 'components';
import { inject, observer } from 'mobx-react';
import PropTypes from 'prop-types';


// import UpdatePortfolioModal from '../../components/Modal/UpdatePortfolio';
import RegularButton from '../../components/CustomButtons/Button';
import CreatePortfolio from '../../components/Modal/CreatePortfolio';
import PortfoliosTable from '../../components/CustomTables/PortfoliosTable';
import IntegrationsTable from '../../components/CustomTables/IntegrationsTable';

import AddApiAccount from '../../components/Modal/ApiAccountModals/AddApiAccount';

// import IconButton from '../../components/CustomButtons/IconButton';
// import CreatePortfolio from '../../components/Modal/CreatePortfolio';

@inject('MarketStore')
@observer
class Settings extends Component {
  state = {
    br: false,
  };

  getMarketSummaries = () => {
    this.props.MarketStore.getSyncedSummaries();
    this.props.MarketStore.getBaseCurrencies();
  };

  showNotification(place) {
    const x = [];
    x[place] = true;
    this.setState(x);
    setTimeout(() => {
      x[place] = false;
      this.setState(x);
    }, 6000);
  }

  render() {
    return (
      <Grid container>
        <ItemGrid xs={12} sm={12} md={12}>
          <RegularCard
            cardTitle="API Integrations"
            button={<AddApiAccount />}
            content={
              <IntegrationsTable
                tableHead={['Exchange', 'Status', 'EDIT', 'DELETE']}
                tableData={[['Poloniex', 'Inactive', '', '']]}
              />
            }
          />
        </ItemGrid>
        <ItemGrid xs={12} sm={12} md={12}>
          <RegularCard
            cardTitle="Portfolios"
            button={<CreatePortfolio />}
            content={
              <PortfoliosTable
                tableHead={[
                  'Name',
                  'Number of Shares',
                  'Current share price',
                  'Total Amount',
                  '',
                ]}
                // tableData={[
                //   ["Poloniex", "Inactive", "test", "test", ""],
                //   ["Poloniex", "Inactive", "test", "test", ""],
                //   ["Poloniex", "Inactive", "test", "test", ""],
                //   ["Poloniex", "Inactive", "test", "test", ""]
                // ]}
              />
            }
          />
        </ItemGrid>

        <Grid container>
          <ItemGrid xs={12} sm={3} md={3}>
            <RegularButton color="primary" onClick={this.getMarketSummaries}>
              Get Markets
            </RegularButton>
          </ItemGrid>

          <ItemGrid xs={12} sm={3} md={3}>
            <Button
              fullWidth
              color="primary"
              onClick={() => this.showNotification('br')}
            >
              Bottom Right Notification
            </Button>

            <Snackbar
              place="br"
              color="info"
              message="This is a warning Message"
              open={this.state.br}
              closeNotification={() => this.setState({ br: false })}
              close
            />
          </ItemGrid>
        </Grid>
      </Grid>
    );
  }
}

Settings.propTypes = {
  MarketStore: PropTypes.object,
};

export default Settings;
