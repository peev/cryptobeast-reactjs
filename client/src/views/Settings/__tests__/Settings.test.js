import React from 'react';
import { shallow } from 'enzyme';
import Settings from './../Settings';

describe('Views/Settings.jsx', () => {
  const settings = shallow(<Settings />);

  it('should render without crashing', () => {
    expect(settings).toMatchSnapshot();
  });
});
