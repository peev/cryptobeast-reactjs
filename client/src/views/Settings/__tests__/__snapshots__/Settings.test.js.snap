// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Views/Settings.jsx should render without crashing 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <inject-_class-with-MarketStore
    MarketStore={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "MarketStore": [Function],
    },
    "ref": [Function],
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "MarketStore": [Function],
      },
      "ref": [Function],
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;
